src/clone.d

```diff
@@ -240,16 +239,67 @@ extern (C++) FuncDeclaration buildOpAssign(StructDeclaration sd, Scope* sc)
     auto fop = new FuncDeclaration(declLoc, Loc(), Id.assign, stc, tf);
     fop.storage_class |= STCinference;
     fop.generated = true;
-    Expression e = null;
-    if (stc & STCdisable)
+
+    if (!(stc & STCdisable))
+        fop.fbody = buildOpAssignBody(sd);
+
+    sd.members.push(fop);
+    fop.addMember(sc, sd);
+    sd.hasIdentityAssign = true; // temporary mark identity assignable
+    uint errors = global.startGagging(); // Do not report errors, even if the
+    Scope* sc2 = sc.push();
+    sc2.stc = 0;
+    sc2.linkage = LINKd;
+    fop.semantic(sc2);
+    fop.semantic2(sc2);
+    // Bugzilla 15044: fop.semantic3 isn't run here for lazy forward reference resolution.
+
+    sc2.pop();
+    if (global.endGagging(errors)) // if errors happened
     {
+        // Disable generated opAssign, because some members forbid identity assignment.
+        fop.storage_class |= STCdisable;
+        fop.fbody = null; // remove fbody which contains the error
     }
-    else if (sd.dtor || sd.postblit)
+
+    //printf("-StructDeclaration::buildOpAssign() %s, errors = %d, disabled = %d\n", sd.toChars(), errors, (fop.storage_class & STCdisable) != 0);
+    return fop;
+}
+
+/**
+ * Build the generated opAssign's body
+ *
+ * The `sd` struct is the "target" struct, opAssign's parameter is the "source".
+ * The generated `opAssign` is either a memberwise copy or a blit of the source
+ * over the target.
+ * TODO: Why ?
```

Comment:
The actual question why we don't do the blit in both situation ? Since we can do it for `struct` with a destructor or postblit, why do we do it memberwise where there is neither ?